### PR TITLE
Fix: comparing double with long 

### PR DESCRIPTION
**Pull request by milaGGL**
SDK sorts cross-type numbers differently than the backend in some edge cases, ie, overflowed double numbers with long.MAX_VALUE. 
Introduce more conditional comparison for double and long numbers.

### CODE REVIEW

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/OrderTest.java [hunk 1]
```diff
@@ -42,88 +42,91 @@ public void verifyOrder() {
...
+    groups[63] = new Value[] {objectValue("bar", intValue(0), "foo", intValue(1))};
+    groups[64] = new Value[] {objectValue("bar", intValue(1))};
+    groups[65] = new Value[] {objectValue("bar", intValue(2))};
+    groups[66] = new Value[] {objectValue("bar", stringValue("0"))};
 
     for (int left = 0; left < groups.length; left++) {
       for (int right = 0; right < groups.length; right++) {
         for (int i = 0; i < groups[left].length; i++) {
           for (int j = 0; j < groups[right].length; j++) {
+            System.out.println(groups[left][i] + "    " + groups[right][j]);
```
Comment (ehsannas) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/OrderTest.java:129]: Remove `System.out.println`

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java [hunk 2]
```diff
@@ -28,12 +28,8 @@
 import com.google.cloud.firestore.*;
 import com.google.cloud.firestore.Query.Direction;
 import java.time.Duration;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (ehsannas) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java:31]: The Java style guide [disallows](https://google.github.io/styleguide/javaguide.html#s3.3.1-wildcard-imports) wildcard imports (`import foo.*`). Although sadly this hasn't been respected in a few places (such as line 28 above).

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/OrderTest.java (hunk 1 above)
Comment (milaGGL) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/OrderTest.java:129]: oops

**Review by ehsannas — APPROVED** [review]
